lib/elixir/test/elixir/calendar/datetime_test.exs

```diff
@@ -287,12 +287,151 @@ defmodule DateTimeTest do
              }
   end
 
+  test "from_iso8601/3 with basic format handles positive and negative offsets" do
+    assert DateTime.from_iso8601("20150124T095007-1000", Calendar.ISO, :basic) |> elem(1) ==
+             %DateTime{
+               microsecond: {0, 0},
+               month: 1,
+               std_offset: 0,
+               time_zone: "Etc/UTC",
+               utc_offset: 0,
+               year: 2015,
+               zone_abbr: "UTC",
+               day: 24,
+               hour: 19,
+               minute: 50,
+               second: 7
+             }
+
+    assert DateTime.from_iso8601("20150124T095007+1000", Calendar.ISO, :basic) |> elem(1) ==
+             %DateTime{
+               microsecond: {0, 0},
+               month: 1,
+               std_offset: 0,
+               time_zone: "Etc/UTC",
+               utc_offset: 0,
+               year: 2015,
+               zone_abbr: "UTC",
+               day: 23,
+               hour: 23,
+               minute: 50,
+               second: 7
+             }
+
+    assert DateTime.from_iso8601("00000101T012207+1030", Calendar.ISO, :basic) |> elem(1) ==
+             %DateTime{
+               microsecond: {0, 0},
+               month: 12,
+               std_offset: 0,
+               time_zone: "Etc/UTC",
+               utc_offset: 0,
+               year: -1,
+               zone_abbr: "UTC",
+               day: 31,
+               hour: 14,
+               minute: 52,
+               second: 7
+             }
+  end
+
+  test "from_iso8601/3 with basic format handles negative dates" do
+    assert DateTime.from_iso8601("-20150124T095007-1000", Calendar.ISO, :basic) |> elem(1) ==
+             %DateTime{
+               microsecond: {0, 0},
+               month: 1,
+               std_offset: 0,
+               time_zone: "Etc/UTC",
+               utc_offset: 0,
+               year: -2015,
+               zone_abbr: "UTC",
+               day: 24,
+               hour: 19,
+               minute: 50,
+               second: 7
+             }
+
+    assert DateTime.from_iso8601("-20150124T095007+1000", Calendar.ISO, :basic) |> elem(1) ==
+             %DateTime{
+               microsecond: {0, 0},
+               month: 1,
+               std_offset: 0,
+               time_zone: "Etc/UTC",
+               utc_offset: 0,
+               year: -2015,
+               zone_abbr: "UTC",
+               day: 23,
+               hour: 23,
+               minute: 50,
+               second: 7
+             }
+
+    assert DateTime.from_iso8601("-00010101T012207+1030", Calendar.ISO, :basic) |> elem(1) ==
```

Comment:
I think you can simplify those tests by comparing the result of the basic with the non-basic. Something like:

```suggestion
    assert DateTime.from_iso8601("-00010101T012207+1030", Calendar.ISO, :basic) |> elem(1) ==
             DateTime.from_iso8601("-0001-01-01T01:22:07+1030", Calendar.ISO)
```